pkg/peertls/extensions.go

```diff
@@ -235,46 +234,40 @@ func (e ExtensionHandlers) VerifyFunc() PeerCertVerificationFunc {
 func (r Revocation) Verify(signingCert *x509.Certificate) error {
 	pubKey, ok := signingCert.PublicKey.(crypto.PublicKey)
 	if !ok {
-		return ErrUnsupportedKey.New("%T", signingCert.PublicKey)
-	}
-
-	data, err := r.TBSBytes()
-	if err != nil {
-		return err
+		return pkcrypto.ErrUnsupportedKey.New("%T", signingCert.PublicKey)
 	}
 
-	if err := VerifySignature(r.Signature, data, pubKey); err != nil {
+	data := r.TBSBytes()
+	if err := pkcrypto.VerifySignature(r.Signature, data, pubKey); err != nil {
 		return err
 	}
 	return nil
 }
 
 // TBSBytes (ToBeSigned) returns the hash of the revoked certificate hash and
 // the timestamp (i.e. hash(hash(cert bytes) + timestamp)).
-func (r *Revocation) TBSBytes() ([]byte, error) {
-	toHash := new(bytes.Buffer)
+func (r *Revocation) TBSBytes() []byte {
```

Comment:
This code in this func looks wrong.

`bytes.Buffer` seems unnecessary, we can use `[]byte` directly and avoid the panic.

`binary.PutVarint` writes the content to the beginning of byte array and does not append.

Reply:
dang.. I can't believe I did this :man_facepalming: